core/src/main/java/nl/nn/adapterframework/jms/JmsTransactionalStorage.java

```diff
@@ -100,20 +102,24 @@ public boolean containsMessageId(String originalMessageId) throws ListenerExcept
 	}
 
 	@Override
-	public S browseMessage(String storageKey) throws ListenerException {
+	public RawMessageWrapper<S> browseMessage(String storageKey) throws ListenerException {
 		try {
 			ObjectMessage msg=browseJmsMessage(storageKey);
-			return (S) msg.getObject();
+			RawMessageWrapper<S> messageWrapper = new RawMessageWrapper<>((S)msg.getObject(), storageKey, null);
+			messageWrapper.getContext().put(PipeLineSession.STORAGE_KEY_KEY, storageKey);
```

Comment:
Ik dacht dat dit een JDBC key key was, maar nu ook voor JMS, misschien dat je hier een handigere naam voor kan bedenken (en/of een eigen setter kan maken in MessageContext)

Reply:
Is nu `STORAGE_ID_KEY`, is dat een betere naam?